using_images/s2i_images/java.adoc

```diff
@@ -115,89 +128,478 @@ or from binary artifacts.
+|`extras/*`
+
+|`*S2I_SOURCE_CONFIGURATION_DIR*`
+|Relative path to directory containing application configuration files to be copied over to the product configuration directory, see `*S2I_TARGET_CONFIGURATION_DIR*`.  Defaults to `configuration`.
+|`configuration`
+
+|`*S2I_SOURCE_DATA_DIR*`
+|Relative path to directory containing application data files to be copied over to the product data directory, see `*S2I_TARGET_DATA_DIR*`. Defaults to `data`.
+|`data`
+
+|`*S2I_SOURCE_DEPLOYMENTS_DIR*`
+|Relative path to directory containing binary files to be copied over to the product deployment directory, see `*S2I_TARGET_DEPLOYMENTS_DIR*`. Defaults to `deployments`.
+|`deployments`
+
+|`*S2I_SOURCE_DIR*`
+|Location of mount for source code to be built. This should not be overridden by end users.
+|`${S2I_DESTINATION_DIR}/src}`
+
+|`*S2I_TARGET_CONFIGURATION_DIR*`
+|Absolute path to which files located in `*S2I_SOURCE_DIR*`/`*S2I_SOURCE_CONFIGURATION_DIR*` are copied.
+|`/opt/eap/standalone/configuration`
+
+|`*S2I_TARGET_DATA_DIR*`
+|Absolute path to which files located in `*S2I_SOURCE_DIR*`/`*S2I_SOURCE_DATA_DIR*` are copied.
+|`/opt/eap/standalone/data`
+
+|`*S2I_TARGET_DEPLOYMENTS_DIR*`
+|Absolute path to which files located in `*S2I_SOURCE_DIR*`/`*S2I_SOURCE_DEPLOYMENTS_DIR*` are copied. Additionally, this is the directory to which build output is copied.
+|`/deployments`
+
+|`*http_proxy*`
+|The location of the http proxy. This takes precedence over `*HTTP_PROXY*` and will be used for both Maven builds and Java runtime.
+|`\http://127.0.0.1:8080`
+
+|`*https_proxy*`
+|The location of the https proxy. This takes precedence over `*HTTPS_PROXY*`, `*http_proxy*`, and `*HTTP_PROXY*`, and will be used for both Maven builds and Java runtime.
+|`myuser:mypass@127.0.0.1:8080`
+
+|`*no_proxy*`
+|A comma separated list of hosts, IP addresses or domains that can be accessed directly. This takes precedence over `*NO_PROXY*` and will be used for both Maven builds and Java runtime.
+|`*.example.com`
+
+|`*prefix_MAVEN_MIRROR_ID*`
+|ID to be used for the specified mirror.  If omitted, a unique ID will be generated.
+|`internal-mirror`
+
+|`*prefix_MAVEN_MIRROR_OF*`
+|Repository IDs mirrored by this entry.  Defaults to `external:`.
+|-
+
+|`*prefix_MAVEN_MIRROR_URL*`
+|The URL of the mirror.
+|`\http://10.0.0.1:8080/repository/internal`
+
+|`*prefix_MAVEN_REPO_DIRECTORY_PERMISSIONS*`
+|Maven repository directory permissions.
+|`775`
+
+|`*prefix_MAVEN_REPO_FILE_PERMISSIONS*`
+|Maven repository file permissions.
+|`664`
+
+|`*prefix_MAVEN_REPO_HOST*`
+|Maven repository host (if not using fully defined url; will fallback to service).
```

Comment:
Suggested edit: 

Maven repository host, if not using fully defined URL, falls back to service.